x/profile/keeper/msg_server_delete_coordinator.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/spn/x/profile/types"
+)
+
+func (k msgServer) DeleteCoordinator(
+	goCtx context.Context,
+	msg *types.MsgDeleteCoordinator,
+) (*types.MsgDeleteCoordinatorResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Check if the coordinator address is already in the store
+	coordByAddress, found := k.GetCoordinatorByAddress(ctx, msg.Address)
+	if !found {
+		return &types.MsgDeleteCoordinatorResponse{},
+			sdkerrors.Wrap(types.ErrCoordAddressNotFound, msg.Address)
+	}
+
+	coord := k.GetCoordinator(ctx, coordByAddress.CoordinatorId)
+	if (coord == types.Coordinator{}) {
+		panic("a coordinator address is associated to a non-existent coordinator ID")
```

Comment:
I think we should return with an error instead. Otherwise, clients wouldn't know what actually went wrong because the content of panic will be kept secret.

Reply:
We use `panic` because this is part of code that should never be reached under any condition. If we reach this, this mean there is a serious inconsistency issue in the chain and it must then be stopped.

This is what is done with Cosmos SDK for this kind of case. You have lot of examples in this file for ex: https://github.com/cosmos/cosmos-sdk/blob/760bde3fbd0b3811a3b7a9ee7d4431544662d5dd/x/bank/keeper/keeper.go#L283

Reply:
If this case is never reached we don't need to handle it. But if it somehow reaches, then we should let the client know so it can know what actually did went wrong.

Reply:
We can't return an error otherwise the blockchain continue to run.

The whole network must be stopped if we reach that point

Reply:
What do you mean by whole network needs to be stopped? You mean nodes' process should be stopped or returning with an error is not determenistic and cannot be reproduced among other nodes?

Reply:
Any node that receives the message will be stopped therefore the blockchain will be halted

Reply:
No, I don't think that's the case. Chains won't get halted when you panic inside the message handler because it's executed in seperate goroutine and later recovered before returning the tx result to the client. Halting it would make the chain vulnerable to attacks anyway as this opens the opportunity of taking the whole network down.

So, I think it's better to just return an error to make the client experience better, panic'ing won't help with anything as it obscures the underlying the error message.

Reply:
> Halting it would make the chain vulnerable to attacks anyway as this opens the opportunity of taking the whole network down.

This is the point, the network must go down if we reach this point

This is the methodology used by Cosmos SDK so far. Another example is the `crisis` module, specifically created to halt the network if an invariant is broken. The `VerifyInvariant` will `panic` if an invariant in the chain is broken: https://github.com/cosmos/cosmos-sdk/blob/760bde3fbd0b3811a3b7a9ee7d4431544662d5dd/x/crisis/keeper/msg_server.go#L50
We must keep synced with Cosmos SDK methodology for this and not implement our own method. If they eventually update this, we must then follow them.

But we must never let the network alive if a severe inconsistency is detected

Reply:
I'm no sure how crisis module halts the chain actually or if halting is relatated to panic'ing inside a message handler. As I checked, chain does not stop and continues to generate blocks even if you panic inside a handler. 

Also, error message inside this panic is never sent to the client so, this is why I suggest to use errors instead. If you try this yourself and panic inside a handler, you'll see that panic's message is obscured. Maybe it's only printed in serverside if you enabled debug mode.

Reply:
Yes, the blockchain actually continue to run. This is not the behavior I was expected to. Thanks for having checked that.

I need to raise the question with engineering, maybe halting chain can only occur through invariants.

Reply:
Let's try to settle this for now.

What about implementing a method:
```
func CritialError(message) error {
  return sdkerrors.Wrap(ErrCritical, message)
}
```

That actually returns a message `CRITICAL: <message>` from the handler

So that whatever the solution we gonna use to handle crisis situation, we can later modify this function

Reply:
Looks good to me.